File: src/ProjectR/content/scripts/spells/LetsSpeedThisUp.lua
function GetTargetType()
    return Allies
end

function GetName()
  return "Let's speed this up"
end

function GetDescription()
  return "Instantly fills the speed bar of your allies to 100%%."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 2
end

function GetDelay()
  return 0
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  if not IsSameChar(a, d) then
    d:SetTurnCounter(GetTimeToAction())
  end
end

File: src/ProjectR.Scripting/content/scripts/spells/LetsSpeedThisUp.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class LetsSpeedThisUp : SpellScriptBase
    {
        public override string Name { get { return "Let's Speed This Up!"; } }
        public override string Description { get { return "Instantly fills the speed bar of your allies to 100%%."; } }

        public override TargetType TargetType { get { return TargetType.Allies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 200; } }
        public override double Delay { get { return 0; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            if (target != caster)
            {
                target.TurnCounter = target.TimeToAction;
            }
        }
    }
}